odoo/addons/base/models/res_lang.py

```diff
@@ -6,17 +6,50 @@
 import locale
 import logging
 import re
-from operator import itemgetter
+from typing import Any, Literal
 
 from odoo import api, fields, models, tools, _
 from odoo.exceptions import UserError, ValidationError
+from odoo.tools import OrderedSet
 
 _logger = logging.getLogger(__name__)
 
 DEFAULT_DATE_FORMAT = '%m/%d/%Y'
 DEFAULT_TIME_FORMAT = '%H:%M:%S'
 
 
+class LangData(tools.frozendict):
+    """ A data class which can access field value like a res.lang record
+    e.g. lang_data.code
+    """
+    __slots__ = ()
+
+    def __getattribute__(self, field: str) -> Any:
+        try:
+            return super().__getitem__(field)
+        except KeyError:
+            if self:
+                return object.__getattribute__(self, field)
+            return False
+
+    def __getitem__(self, field: str) -> Any:
+        try:
+            return super().__getitem__(field)
+        except KeyError:
+            if self:
+                raise
+            return False
```

Comment:
Tricky use case 😞 
Using `lang_data.keys()` for active language is not legit but works like a dict.
`lang_data.keys()` for dummy langdata will raise error since `lang_data.keys is False`